### ISSUE TITLE
pact-jvm-provider-gradle fails to verify a message provider for versions 3.2.5, 3.2.6 and 3.2.7

### ISSUE THREAD

**Issue by BenSayers** [DiUS/pact-jvm#259]
While trying to use the pact-jvm-provider-gradle plugin (version 3.2.5, 3.2.6 or 3.2.7) to verify a message provider I encounter the following error:
```
Failures:

0) Verifying a pact between customer-details-consumer and customer-details-provider - a created event for the new customer Given a new customer is created
      No signature of method: static au.com.dius.pact.provider.ResponseComparison.compareMessage() is applicable for argument types: (au.com.dius.pact.model.v3.messaging.Message, java.lang.String) values: [au.com.dius.pact.model.v3.messaging.Message(a created event for the new customer, a new customer is created, au.com.dius.pact.model.OptionalBody(PRESENT, {"address":{"addressLine1":"address-line-1","addressLine2":"address-line-2","personId":1},"person":{"firstName":"Mr","id":1,"lastName":"Test"}}), [:], [Content-Type:application/json; charset=UTF-8, contentType:application/json]), ...]
      Possible solutions: compareMessage(au.com.dius.pact.model.v3.messaging.Message, au.com.dius.pact.model.OptionalBody)
```

When I downgrade to version 3.2.4 the same test passes:
```
  Given a new customer is created
         WARNING: State Change ignored as there is no stateChange URL
  a created event for the new customer
    generates a message which
      has a matching body (OK)

BUILD SUCCESSFUL
```

While I don't think my pact file or provider test function contain anything special I have provided them below in case it helps with reproducing the issue.

Pact file
```javascript
{
    "consumer": {
        "name": "customer-details-consumer"
    },
    "provider": {
        "name": "customer-details-provider"
    },
    "messages": [
        {
            "metaData": {
                "contentType": "application/json",
                "Content-Type": "application/json; charset=UTF-8"
            },
            "providerState": "a new customer is created",
            "contentType": "application/json",
            "contents": {
                "address": {
                    "addressLine1": "address-line-1",
                    "addressLine2": "address-line-2",
                    "personId": 1
                },
                "person": {
                    "firstName": "Mr",
                    "id": 1,
                    "lastName": "Test"
                }
            },
            "matchingRules": {
                
            },
            "description": "a created event for the new customer"
        }
    ],
    "metadata": {
        "pact-specification": {
            "version": "3.0.0"
        },
        "pact-jvm": {
            "version": "3.2.7"
        }
    }
}
```

Provider function
```java
    @PactVerifyProvider("a created event for the new customer")
    public String shouldCreateANewCustomer() {
        CustomerDetails customerDetails = new CustomerDetails(
                new Address(1, "address-line-1", "address-line-2"),
                new Person(1, "Mr", "Test")
        );

        return JsonOutput.toJson(customerDetails);
    }
```

**Comment by uglyog** [comment]
It must be a regression. I'll have a look at what has changed since 3.2.4.

**Comment by uglyog** [comment]
Fixed now

**Comment by uglyog** [comment]
Released fix with versions 3.2.8/2.4.10
